bluer-tools/src/blumon.rs

```diff
@@ -76,13 +160,85 @@ impl DeviceMonitor {
                         } else {
                             self.show_device(data).await;
                         }
+
+                        let device = self.adapter.device(data.address)?;
+
+                        // Assuming `device.manufacturer_data().await?` returns a `HashMap<u16, Vec<u8>>`
+                        let manufacturer_data_map = device.manufacturer_data().await?.unwrap_or_default();
+
+                        // Convert the HashMap into a string representation
+                        let manufacturer_data_string = manufacturer_data_map.iter()
+                        .map(|(key, value)| {
+                            // Convert key to hex string with "0x" prefix
+                            let key_hex = format!("0x{:04X}", key);
+                    
+                            // Convert each byte in the vector to its hex representation with "0x" prefix
+                            // If the byte is a printable ASCII character, also show the character in parentheses
+                            let value_hex = value.iter()
+                                .map(|byte| {
+                                    let hex_str = format!("0x{:02X}", byte);
+                                    if byte.is_ascii_graphic() || *byte == b' ' { // Includes space as printable
+                                        format!("{}({})", hex_str, *byte as char)
+                                    } else {
+                                        hex_str
+                                    }
+                                })
+                                .collect::<Vec<String>>()
+                                .join(" ");
+                    
+                            format!("{}: [{}]", key_hex, value_hex)
+                        })
+                        .collect::<Vec<String>>()
+                        .join(", ");
+                    // ***********************************************************************
+                        // Assuming device.service_data().await?.unwrap_or_default().to_string(),
+                        let service_data_map = device.service_data().await?.unwrap_or_default();
+
+                        // Convert the HashMap into a string representation
+                        let service_data_string = service_data_map.iter()
```

Comment:
Same as above. Key can be UUID as string but value can just stay as-is, i.e. array of numbers.

Reply:
I think the advantage is to improve the usability of the log file by making it easy to use command line tools to search for UUIDs which are given as hex values in the Bluetooth spec and also when using BlueZ tools.

Reply:
Same as above.